autonat.go

```diff
@@ -304,6 +328,17 @@ func (as *AmbientAutoNAT) recordObservation(observation autoNATResult) {
 	}
 }
 
+func (as *AmbientAutoNAT) tryProbe(p peer.ID) bool {
+	info := as.host.Peerstore().PeerInfo(p)
+	// TODO: reject if recently probed / on backoff
+
+	if !as.config.dialPolicy.skipPeer(info.Addrs) {
```

Comment:
@willscott Even though the docs for `skipDial` say that the client might use them when considering whether to dial a specific server address, we don't actually use it anywhere. Can we change the docs there or is that a mistake ?

Reply:
clarified comment